sdk/communication/communication-rooms/README.md

```diff
@@ -18,24 +18,189 @@ Communication Rooms Client
+
+## Examples
+
+## Authentication
+
+You can get a key and/or connection string from your Communication Services resource in [Azure Portal][azure_portal]. Once you have a key, you can authenticate the `RoomsClient` with any of the following methods:
+
+### Create `KeyCredential` with `AzureKeyCredential` before initializing the client
+
+```typescript
+import { AzureKeyCredential } from "@azure/core-auth";
+import { RoomsClient } from "@azure/communication-rooms";
+
+const credential = new AzureKeyCredential(KEY);
+const client = new RoomsClient(ENDPOINT, credential);
+```
+
+### Using a connection string
+
+```typescript
+import { RoomsClient } from "@azure/communication-rooms";
+
+const connectionString = `endpoint=ENDPOINT;accessKey=KEY`;
+const client = new RoomsClient(connectionString);
+```
+
+### Using a `TokenCredential`
+
+```typescript
+import { DefaultAzureCredential } from "@azure/identity";
+import { RoomsClient } from "@azure/communication-rooms";
+
+const credential = new DefaultAzureCredential();
+const client = new RoomsClient(ENDPOINT, credential);
+```
+
+If you use a key to initialize the client you will also need to provide the appropriate endpoint. You can get this endpoint from your Communication Services resource in [Azure Portal][azure_portal].
+
+## Usage
+
+### Create a room
+
+To create a room, call the `createRoom` method. All settings are optional.
+
+If `validFrom` is not provided, it is defaulted to the current datetime. If `validUntil` is not provided, the default is `validFrom + 180 days`.
+
+When defining `participants`, if `role` is not specified, then it will be `attendee` by default.
+
+```js
+// create users with CommunicationIdentityClient
+const identityClient = new CommunicationIdentityClient(connectionString);
+const user1 = await identityClient.createUserAndToken(["voip"]);
+
+// create RoomsClient
+const roomsClient: RoomsClient = new RoomsClient(CONNECTION_STRING);
+
+const validFrom = new Date(Date.now());
+let validForDays = 10;
+let validUntil = validFrom.setDate(validFrom.getTime() + validForDays);
+
+// options payload to create a room
+const createRoomOptions: CreateRoomOptions = {
+  validFrom,
+  validUntil,
+  participants: [
+    {
+      id: user1.user,
+      role: "Attendee",
+    },
+  ],
+};
+
+// create room
+const room = await roomsClient.createRoom(createRoomOptions);
+```
+
+[Find CommunicationIdentityClient here](https://github.com/Azure/azure-sdk-for-js/edit/main/sdk/communication/communication-identity)
+
+### Update a room
+
+To update the `validFrom` and `validUntil` settings of a room use the `updateRoom` method.
+
+```js
+validForDays = 60;
+validUntil = validFrom.setDate(validFrom.getDate() + validForDays);
+const updateRoomOptions: UpdateRoomOptions = {
```

Comment:
If we just update validUntil, the service will complain. Both validFrom and validUntil needs to be set, so let's make sure  that is reflected in the README example